homeassistant/components/esphome/climate.py

```diff
@@ -166,12 +187,20 @@ def fan_modes(self):
         return [
             _fan_modes.from_esphome(mode)
             for mode in self._static_info.supported_fan_modes
-        ]
+        ] + list(self._static_info.supported_custom_fan_modes)
 
     @property
     def preset_modes(self):
         """Return preset modes."""
-        return [PRESET_AWAY, PRESET_HOME] if self._static_info.supports_away else []
+        if self._static_info.supports_away:
+            return [PRESET_AWAY, PRESET_HOME]
```

Comment:
You can change the `elif:` to `if:` to fix this error. `else: return` can be replaced by a basic `return`. So:
```
if:
  return
if:
  return
return
```

Reply:
Fixed, thanks